### PR TITLE
Register v3 server in integration tests

### PR DESCRIPTION
**Pull request by jyotimahapatra**
This PR registers the integration tests with v3 grpc and rest endpoints.

Future PR:
1. change `integration.sh` and bootstrap files to add v3 specific integration tests.

### CODE REVIEW

**Comment by jyotimahapatra** [comment]
@kyessenov PTAL

**Comment by jyotimahapatra** [comment]
@kyessenov This PR registers the v3 grpc server with the integration test binary. Next PR will change the bootstrap config to really exercise the v3 code path in the tests.

**Comment by jyotimahapatra** [comment]
@kyessenov Could you take a look?

File: pkg/test/v2/register.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// RegisterAccessLogServer starts an accessloggrpc service.
+func RegisterAccessLogServer(grpcServer *grpc.Server, als *AccessLogService) {
+	accessloggrpc.RegisterAccessLogServiceServer(grpcServer, als)
+}
+
+// RegisterServer registers with v2 services.
+func RegisterServer(grpcServer *grpc.Server, server server.Server) {
+	// register services
+	discoverygrpc.RegisterAggregatedDiscoveryServiceServer(grpcServer, server)
+	endpointservice.RegisterEndpointDiscoveryServiceServer(grpcServer, server)
```
Comment (phylake) [inline pkg/test/v2/register.go:41]: @alecholmez why is ADS and individual xDS registered? The interface passed to `RegisterAggregatedDiscoveryServiceServer` must implement `DeltaAggregatedResources(AggregatedDiscoveryService_DeltaAggregatedResourcesServer) error`. The `AggregatedDiscoveryService_DeltaAggregatedResourcesServer` can send/recv DeltaDiscovery(Response|Request). Trying to figure out what [variant(s)](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#four-variants) of xDS I'm looking at here. All 4?

Still pouring through the source just hoping for a quicker answer. Thanks

**Pull request merged**